pydantic/decorators.py

```diff
@@ -85,26 +182,31 @@ def root_validator(
     before or after standard model parsing/validation is performed.
     """
     if __func:
-        f_cls = _decorators.prepare_decorator(__func, allow_reuse)
+        f_cls = _decorators.prepare_validator_decorator(__func, allow_reuse)
         setattr(f_cls, _decorators.ROOT_VALIDATOR_TAG, _decorators.Validator(mode=mode))
         return f_cls
 
     def dec(f: Callable[..., Any]) -> classmethod[Any]:
-        f_cls = _decorators.prepare_decorator(f, allow_reuse)
+        f_cls = _decorators.prepare_validator_decorator(f, allow_reuse)
         setattr(f_cls, _decorators.ROOT_VALIDATOR_TAG, _decorators.Validator(mode=mode))
         return f_cls
 
     return dec
 
 
+_SerializerFunctionType = TypeVar(
+    '_SerializerFunctionType', bound=Union[_core_schema.SerializeWrapFunction, _core_schema.SerializePlainFunction]
```

Comment:
Important to note: I am explicitly prohibiting instance method serializers, at least for now, since they’re broken and if we want to do that right I think we need to rethink some things